testing/src/test/java/org/oppia/testing/CoroutineExecutorServiceTest.kt

```diff
@@ -0,0 +1,1138 @@
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.never
+import org.mockito.Mockito.verify
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.threading.BackgroundDispatcher
+import org.robolectric.annotation.LooperMode
+import java.util.concurrent.Callable
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.Executor
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.RejectedExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+
+/**
+ * Tests for [CoroutineExecutorService]. NOTE: significant care should be taken when modifying these
+ * tests since they combine several different coroutine dispatchers, including a real-time
+ * dispatcher, the ones coordinated by [TestCoroutineDispatchers], and custom test dispatchers in
+ * order to test complex scenarios like whether certain blocking operations block/timeout/complete
+ * as expected.
+ *
+ * Many of these tests also depend on real time, which means different performing machines may
+ * introduce flakes when running these tests. Please reach out to oppia-android-dev@googlegroups.com
+ * if you find yourself in this situation.
+ *
+ * For developers changing this suite: note that an n-threaded real dispatcher is used to test
+ * blocking operations since coordinating multiple co-dependent test dispatchers creates a circular
+ * dependency that effectively always results in a deadlock (since there's no way to control the
+ * order of tasks being executed in a coroutine dispatcher).
+ */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class CoroutineExecutorServiceTest {
+  @Rule
+  @JvmField
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Inject
+  @field:BackgroundDispatcher
+  lateinit var backgroundCoroutineDispatcher: CoroutineDispatcher
+
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
```

Comment:
we can reformat this. 

```
  @Inject
  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers

  @Inject
  lateinit var testDispatcherFactory: TestCoroutineDispatcher.Factory

  @Mock
  lateinit var mockRunnable: Runnable

  @Mock
  lateinit var mockCallable: Callable<String>
```

Reply:
I prefer keeping the current format since it's more concise & still conformant with the linter.

Reply:
Resolving this comment on behalf of @anandwana001 , so as to unblock this PR.